### PR TITLE
feat: Add test failure slack notification

### PR DESCRIPTION
**Pull request by radnov**

### CODE REVIEW

**Review by teleivo — APPROVED** [review]
🥇

File: vars/slack.groovy [hunk 1]
```diff
@@ -1,41 +1,49 @@
...
-    "#ff0000",
+    '#ff0000',
     buildUrl() + " failed."
   )
 }
 
+def sendTestsFailureMessage(channel) {
+  sendMessage(
+    '#ff0000',
+    buildUrl() + "\nLatest test run on ${GIT_BRANCH} failed and needs investigation. :detective-duck:\nCommit: <https://github.com/dhis2/dhis2-core/commit/${GIT_COMMIT}|${GIT_COMMIT}>",
```
Comment (vilkg) [inline vars/slack.groovy:11]: hmm, this sounds very specific to the core repo and not very reusable. Should we maybe have this function accept detailed message to send? Or skip it all together and use sendMessage in [here](https://github.com/dhis2/dhis2-core/pull/11543)?

Comment (teleivo) [reply vars/slack.groovy:11]: sounds like a good idea 👍🏻

Comment (radnov) [reply vars/slack.groovy:11]: What about adding the repository url as an argument instead? The rest of the message is okay I suppose.   `sendTestFailureMessage(channel, repositoryUrl)` 🤔

Comment (vilkg) [reply vars/slack.groovy:11]: Except that GIT_BRANCH env variable is only present in multibranch pipelines and will fail on the regular ones. But might not be a big problem

Comment (radnov) [reply vars/slack.groovy:11]: I'll just go with the simplest and use `sendMessage` directly in the dev pipeline.

**Pull request merged**